kong/core/balancer.lua

```diff
@@ -703,7 +703,8 @@ local function execute(target)
     hostname = target.host
     if not ip then
       log(ERR, "[dns] ", port, ". Tried: ", tostring(try_list))
-      if port == "dns server error: 3 name error" then
+      if port == "dns server error: 3 name error" or
+         port == "dns client error: 101 empty record received" then
```

Comment:
@Tieske A question, as I try to understand the relationship between lua-resty-dns-client and kong.core.balancer better: is there any other error (matching `"dns .* error"`) returned in `port` that should _not_ result in a `name resolution failed` error?

Reply:
Not if everything works as expected. But it could probably return db errors as well if a lookup for an upstream fails for some reason.

So these are the specific ones we catch to generate a 503. Others will be 500. (Top of mind answer, on mobile)

Reply:
Didn't read your question properly. 

So these two errors listed in this change, are the only two where everything works as it should, yet we could not resolve.

So for example a dns server error is now reported as a 500.

Reply:
> So these two errors listed in this change, are the only two where everything works as it should, yet we could not resolve.

To check that I follow: this means that these are the cases where communication with Kong and the DNS server went well, but the name could not be resolved, and the other ones are cases include things like "DNS server is down", "DNS server replied badly formatted message" (these two being 'their' fault), "we couldn't open a socket" (this one being 'our' fault), etc.?

Reply:
yes

Reply:
Got it, thank you for the clarification!